### PR TITLE
fix: subscription sync process stuck after gateway restart 

### PR DESCRIPTION
**Pull request by jgiovaresco**
## Issue

https://gravitee.atlassian.net/browse/APIM-1538

## Description

Start the subscription of the dispatch completable immediately to prevent waiting forever and blocking the remaining steps of the synchronization process.

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-abqodorriy.chromatic.com)

### CODE REVIEW

File: gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/pom.xml [hunk 1]
```diff
@@ -127,6 +127,12 @@
             <artifactId>vertx-web-client</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.awaitility</groupId>
+            <artifactId>awaitility</artifactId>
+            <version>${awaitility.version}</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (phiz71) [inline gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/pom.xml]: Why didn't you add this in the dependencyManagement section of the main pom.xml?
And also I have the feeling that this library could be useful elsewhere in APIM.

Comment (jgiovaresco) [reply gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/pom.xml]: 🤷  It was not done for testcontainers so I supposed I should not 🙈

Comment (phiz71) [reply gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/pom.xml]: Maybe we should also do that for TC 😛

Comment (jgiovaresco) [reply gravitee-apim-gateway/gravitee-apim-gateway-services/gravitee-apim-gateway-services-sync/pom.xml]: a PR is coming

**Review by phiz71 — APPROVED** [review]

**Review by ytvnr — APPROVED** [review]

**Pull request merged**